magda-web-server/src/standardiseUiBaseUrl.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Validate `argv.uiBaseUrl` to make sure it has a leading slash, but no trailing slash
+ *
+ * @export
+ * @param {string} uiBaseUrl
+ * @returns
+ */
+export default function standardiseUiBaseUrl(uiBaseUrl: string) {
+    if (typeof uiBaseUrl !== "string") {
+        return "/";
+    }
+
+    let urlStr = uiBaseUrl.trim();
+    if (!urlStr) {
+        return "/";
+    }
+
+    if (urlStr === "/") {
+        return urlStr;
+    }
+
+    if (urlStr.indexOf("/") !== 0) {
+        urlStr = "/" + urlStr;
+    }
+
+    return urlStr.lastIndexOf("/") === urlStr.length - 1
+        ? urlStr.substr(0, urlStr.length - 1)
+        : urlStr;
+}
```

Comment:
e.g. if base href is set to `/magda/`,

assuming dev.magda.io host,
a link pointing to `/this-route/` will resolve to `https://dev.magda.io/magda/this-route/`

whereas the same link, with href set to `/magda`, will resolve to `https://dev.magda.io/this-route/` because it will strip out the `magda` portion in `/magda`